test/functional/rpc_fundrawtransaction.py

```diff
@@ -1338,6 +1339,68 @@ def test_feerate_rounding(self):
         rawtx = w.createrawtransaction(inputs=[], outputs=[{self.nodes[0].getnewaddress(address_type="bech32"): 1 - 0.00000202}])
         assert_raises_rpc_error(-4, "Insufficient funds", w.fundrawtransaction, rawtx, {"fee_rate": 1.85})
 
+    def test_input_confs_control(self):
+        self.nodes[0].createwallet("minconf")
+        wallet = self.nodes[0].get_wallet_rpc("minconf")
+
+        # Fund the wallet with different chain heights
+        self.nodes[2].sendtoaddress(wallet.getnewaddress(), 1)
+        self.nodes[2].sendtoaddress(wallet.getnewaddress(), 1)
+        self.generate(self.nodes[2], 1)
+        self.nodes[2].sendtoaddress(wallet.getnewaddress(), 1)
+        self.nodes[2].sendtoaddress(wallet.getnewaddress(), 1)
+        self.generate(self.nodes[2], 1)
+
+        self.sync_blocks()
+
+        unconfirmed_txid = wallet.sendtoaddress(wallet.getnewaddress(), 0.5)
+
+        self.log.info("Crafting TX using an unconfirmed input")
+        target_address = self.nodes[2].getnewaddress()
+        raw_tx1 = wallet.createrawtransaction([], {target_address: 0.1}, 0, True)
+        funded_tx1 = wallet.fundrawtransaction(raw_tx1, {'fee_rate': 1, 'maxconf': 0})['hex']
+
+        # Make sure we only had the one input
+        tx1_inputs = self.nodes[0].decoderawtransaction(funded_tx1)['vin']
+        assert_equal(len(tx1_inputs), 1)
+
+        utxo1 = tx1_inputs[0]
+        assert unconfirmed_txid == utxo1['txid']
+
+        final_tx1 = wallet.signrawtransactionwithwallet(funded_tx1)['hex']
+        txid1 = self.nodes[0].sendrawtransaction(final_tx1)
+
+        mempool = self.nodes[0].getrawmempool()
+        assert txid1 in mempool
+
+        self.log.info("Fail to craft a new TX that sends more funds with add_inputs = False")
+        raw_tx2 = wallet.createrawtransaction([{'txid': utxo1['txid'], 'vout': utxo1['vout']}], {target_address: 1})
+        assert_raises_rpc_error(-4, "Insufficient funds", wallet.fundrawtransaction, raw_tx2, {'fee_rate': 1, 'add_inputs': False})
```

Comment:
Any reason for adding this case here?

Seems to not be needed as it's already covered inside the `test_add_inputs_default_value` function and we here want to test the min/max conf params.

Reply:
This commit was authored by @champo before `test_add_inputs_default_value` was added. I have removed this test.